test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -680,15 +906,98 @@ func assertDynamicSecretRotation(t *testing.T, ctx context.Context,
 			}
 			return errs
 		}
+		lastObj = o
 		return nil
 	}, backoff.WithMaxRetries(bo, maxTries),
 	)) {
-		return
+		return nil
 	}
 
 	// check that all rollout-restarts completed successfully
 	if len(vdsObj.Spec.RolloutRestartTargets) > 0 {
 		awaitRolloutRestarts(t, ctx, client,
 			vdsObj, vdsObj.Spec.RolloutRestartTargets)
 	}
+	return &lastObj
+}
+
+func deleteEntitiesBySAPrefix(t *testing.T, vc *api.Client, ctx context.Context, k8sNamespace, saPrefix string) {
+	t.Helper()
+
+	resp, err := vc.Logical().ListWithContext(ctx, "identity/entity/id")
+	if !assert.NoError(t, err, "failed to list identity entities") {
+		return
+	}
+
+	ids, ok := resp.Data["keys"].([]interface{})
+	if !assert.True(t, ok, "keys not found in response") {
+		return
+	}
+
+	var result []string
+	for _, v := range ids {
+		id := v.(string)
+		resp, err := vc.Logical().ReadWithContext(ctx, "identity/entity/id/"+id)
+		if !assert.NoErrorf(t, err, "failed to lookup entity id %s", id) {
+			return
+		}
+
+		aliases, ok := resp.Data["aliases"].([]interface{})
+		if !assert.True(t, ok, "aliases not found in response") {
+			return
+		}
+		for _, alias := range aliases {
+			aliasMap := alias.(map[string]interface{})
+			metadata, ok := aliasMap["metadata"].(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			if v, ok := metadata["service_account_namespace"]; ok && v.(string) == k8sNamespace {
+				if v, ok := metadata["service_account_name"]; ok && strings.HasPrefix(v.(string), saPrefix) {
+					result = append(result, id)
+				}
+			}
+		}
+	}
+
+	if !assert.Greaterf(t, len(result), 0,
```

Comment:
This seems a little hard to grok at first glance. Is it just checking if `len(result) == 0`?